python/_tskitmodule.c

```diff
@@ -4198,6 +4360,44 @@ SiteTable_get_metadata_offset(SiteTable *self, void *closure)
     return ret;
 }
 
+static PyObject *
+SiteTable_get_metadata_schema(SiteTable *self, void *closure)
+{
+    PyObject *ret = NULL;
+
+    if (SiteTable_check_state(self) != 0) {
+        goto out;
+    }
+    ret = PyBytes_FromStringAndSize(self->table->metadata_schema, self->table->metadata_schema_length);
+out:
+    return ret;
+}
+
+static int
+SiteTable_set_metadata_schema(SiteTable *self, PyObject *arg, void *closure)
+{
+    int ret = -1;
+    char *metadata_schema = "";
+    Py_ssize_t metadata_schema_length = 0;
+    
+    if (SiteTable_check_state(self) != 0) {
+        goto out;
+    }
+    if (arg != Py_None) {
+        if (PyBytes_AsStringAndSize(arg, &metadata_schema, &metadata_schema_length) < 0) {
+            goto out;
+        }
+    }
+    ret = tsk_site_table_set_metadata_schema(self->table, metadata_schema, metadata_schema_length);
```

Comment:
I don't think ``tsk_site_table_set_metadata_schema`` returns anything other than 0 on success, so can use the standard ``if (ret != 0)`` and don't bother setting ret = 0

Ah, but we're using ``ret`` as the Python signalling variable. I think it's best to split the concerns here, and use ``ret`` as the variable used for the Python return value and use maybe ``err = tsk_site_table_set_metadata_schema`` to track the return value of tskit API calls.

Reply:
The change to string changed how the string was made so just https://github.com/tskit-dev/tskit/pull/491/commits/d5881fa448ede1273e31612fd1e771ce57075e7d fixed this.